vortex-array/src/array/bool/compute/cast.rs

```diff
@@ -0,0 +1,20 @@
+use vortex_dtype::DType;
+use vortex_error::VortexResult;
+
+use crate::array::{BoolArray, BoolEncoding};
+use crate::compute::CastFn;
+use crate::{Array, IntoArray};
+
+impl CastFn<BoolArray> for BoolEncoding {
+    fn cast(&self, array: &BoolArray, dtype: &DType) -> VortexResult<Array> {
+        assert!(matches!(dtype, DType::Bool(_)));
+
+        // If the types are the same, return the array,
+        // otherwise set the array nullability as the dtype nullability.
+        if array.dtype() != dtype {
+            Ok(BoolArray::new(array.boolean_buffer(), dtype.nullability()).into_array())
```

Comment:
You can't just cast to NonNullable if nulls exist. You need to check `array.all_valid()`

Reply:
Some tests are maybe worth it for this